### PR TITLE
Reader: align options menus on post card and post details

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref #15761

This change does two things:
- The options menu in post details now uses the same as post card. 
- Post details options now has Block and Report when applicable.

Notes: 
- Block and Report where the only differences I found with the two menus. i.e. all the post card and post details options should now match. 
- I have not yet aligned the success/failure messages, so those still differ.
- Following from post details no longer shows a toast message since post card never did. I'll fix this when I align the messaging.
- Just a tip. If you happen to block a site and want to unblock it, https://wordpress.com/me/site-blocks is the place to be.

To test:

Menu change: 
- There should be no visible or functional change. All the menu options should perform the same action as before.

Block / Report:
- Discover > post details now has Block and Report options.
- Discover > You might like > topic > post details now has Report option.
- Following > filter by topic > post details now has Report option.

| Discover post | Discover topic post | Following topic post |
|--------|-------|------|
| discover | discover_topic | followed_topic |

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderShowMenuAction.swift [hunk 1]
```diff
@@ -6,7 +6,13 @@ final class ReaderShowMenuAction {
         isLoggedIn = loggedIn
     }
 
-    func execute(with post: ReaderPost, context: NSManagedObjectContext, topic: ReaderSiteTopic? = nil, readerTopic: ReaderAbstractTopic?, anchor: UIView, vc: UIViewController) {
+    func execute(with post: ReaderPost,
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Reader/ReaderShowMenuAction.swift:9]: Yay for clarifying the argument names. :)

**Review by aerych — COMMENTED** [review]
Howdy @ScoutHarris 👋 

> Post details options now has Block and Report when applicable

I like this change.  Iirc, we excluded the block option from the detail because we figured content that was offensive wouldn't be tapped into.  It's possible that offensive content might not be apparent until clicking into the post, so having the option available makes sense.

>  If you happen to block a site and want to unblock it, https://wordpress.com/me/site-blocks is the place to be.

Thanks :)  I follow the sports tag just for testing site blocks ;)

Question:  If I'm on the post detail and I block a site, should the menu update with the option to unblock that site, or should we automatically dismiss the detail?  Currently the details remains and the menu still says "Block This Site". Tapping the option a second time yields an error.

**Comment by ScoutHarris** [comment]
Hey @aerych .

> It's possible that offensive content might not be apparent until clicking into the post

That. :smile:

>  If I'm on the post detail and I block a site, should the menu update with the option to unblock that site

I'll look into it. 

> should we automatically dismiss the detail?

🤔 I don't think so. That would assume the user meant to tap that option. And since there is no way to unblock sites in the app, it might be better to give them a chance to undo it.

**Comment by aerych** [comment]
> And since there is no way to unblock sites in the app, it might be better to give them a chance to undo it.

Sounds good 👍

**Comment by ScoutHarris** [comment]
Hey @aerych .

> menu still says "Block This Site". Tapping the option a second time yields an error.

The same thing happens on post card, so this just inherited existing behavior. How would you feel about me looking into that separately so I can look at both scenarios and see what's best? And... I'm not quite sure how to fix it yet. 😄

**Comment by aerych** [comment]
Works for me!

**Review by aerych — APPROVED** [review]
:shipit: !

**Comment by ScoutHarris** [comment]
Thanks much @aerych !

**Pull request merged**